kiota_http/_exceptions.py

```diff
@@ -0,0 +1,21 @@
+"""Exceptions raised in Kiota HTTP."""
+
+
+class KiotaHTTPXError(Exception):
+    """Base class for Kiota HTTP exceptions."""
+
+
+class BackingstoreError(KiotaHTTPXError):
```

Comment:
I'm not sure why we're introducing specialized error types here?

Reply:
I added them to get specialized context of the source of the error as well as the kind of the error. Previously we just raised and `Exception(msg)` which is the basic way to do it. However, it limits handling the errors. Specializing the errors for the lib offers:
- ability for the client to catch specific errors. i.e., in tests or when using the library.
- A clear source and kind of error in the stack trace. A `RedirectError` is most probably raised from the `redirect_handler` module and its value will most probably state the kind of redirect error.
- ability to catch _all_ errors under the base error class `KiotaHTTPXError`. So, all the subclassed errors can easily be handled in a single block (if needed by the lib user).

IMO, it's a good practice which goes a long way in developer usage of the lib especially it being a core library.

Reply:
I just saw this comment now, sorry about the late reply. I agree that specialized error types make handling easier in most languages. Couple of challenges here:
- This error is related to the backing store, which is a concept defined in the abstractions, wouldn't it make more sense to have the error defined in abstractions?
- The name has a casing issue
- Python would be the only language doing that to date

Reply:
Oh, I see your point @baywet. For point 1 and 3, the naming is to aid with the raised exception when using the library in Python. A developer will get a custom error name. I believe the experience is different in other languages. Maybe, we can change the name to something more inclined to `Kiota http` like `KiotaHttpBackingStoreError`? For point 2 I don't get the challenge?

Reply:
BackingstoreError => BackingStoreError (note the capital S)